app/models/user.rb

```diff
@@ -0,0 +1,7 @@
+class User < ApplicationRecord
+  has_many :lesson_activities
+  has_many :lessons, through: :lesson_activities
+  has_many :skill_activities
+  has_many :skills, through: :skill_progresses
+  has_and_belongs_to_many :words
```

Comment:
I would suggest you go with has many through instead of has and belongs to many

http://cobwwweb.com/why-i-dont-use-has-and-belongs-to-many-in-rails

http://blog.flatironschool.com/why-you-dont-need-has-and-belongs-to-many/